src/psyclone/psyir/tools/call_tree_utils.py

```diff
@@ -192,7 +192,8 @@ def get_input_parameters(self, read_write_info, node_list,
             # parameter and does not need to be saved. Note that loop variables
             # have a WRITE before a READ access, so they will be ignored
             # automatically.
-            if not variables_info[signature].is_written_first():
+            if (not variables_info[signature].is_written_first() and
```

Comment:
First, instead of being "not" and adding the not is_called here, maybe this should be "is_read_first" and keep all the AccessType logic inside the method? It would also match what the docstring here says.

But in addition, I checked if "is_write_first" considers the UNKNON AccessType as a worst-case scenario for the first access (as I wondered this in a previous comment) and it does not. As this method are currently implemented they mean "is guaranteed to be read/write first?" but I think what we need here is "is potentially read/write first?" (but I am unsure I have not used the CallTreeUtils myself yet)

Maybe this is the typical issue with boolean methods that can be unknow, as we could add the `is_read/write_first(unknown_as=True/False)` to make it explicit that also used in other places?